src/router_core/connections.c

```diff
@@ -1753,6 +1756,12 @@ static void qdr_link_inbound_detach_CT(qdr_core_t *core, qdr_action_t *action, b
         link->auto_link->state = QDR_AUTO_LINK_STATE_FAILED;
         free(link->auto_link->last_error);
         link->auto_link->last_error = qdr_error_description(error);
+
+        //
+        // The auto link has failed. Periodically retry setting up the auto link until
+        // it succeeds.
+        //
+        qdr_route_retry_auto_link_CT(core, link);
```

Comment:
Same comment here:  At this location in the architecture, the fact that there is a retry is abstracted.  This call should be "auto_link_detached" or similar.  Let the route_control module be responsible for what actions it takes when an auto-link is detached by the peer.